### PR TITLE
Updates to some flaky tests

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: src/client/datascience/jupyter/jupyterCellOutputMimeTypeTracker.ts [hunk 1]
```diff
@@ -122,7 +123,14 @@ export class CellOutputMimeTypeTracker implements IExtensionSingleActivationServ
         if (!e.model) {
             return;
         }
-        e.model?.getCellsWithId().forEach(this.checkCell.bind(this));
+        try {
+            e.model?.getCellsWithId().forEach(this.checkCell.bind(this));
+        } catch (ex) {
+            // Can fail on CI, if the notebook has been closed or the like
+            if (!IS_CI_SERVER) {
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/jupyterCellOutputMimeTypeTracker.ts:130]: In one of @DavidKutu PR i noticed this failing.

Comment (DonJayamanne) [reply src/client/datascience/jupyter/jupyterCellOutputMimeTypeTracker.ts:130]: Swallow on CI, but not in end user

File: src/client/datascience/notebookStorage/vscNotebookModel.ts [hunk 2]
```diff
@@ -99,13 +99,15 @@ export class VSCodeNotebookModel extends BaseNotebookModel {
...
         }
-        return this.document.cells.map((cell) => {
-            return {
-                id: cell.document.uri.toString(),
-                data: createJupyterCellFromVSCNotebookCell(cell),
-                state: cellRunStateToCellState(cell.metadata.runState)
-            };
-        });
+        return this.document.cells
+            .filter((cell) => !cell.document.isClosed)
```
Comment (DonJayamanne) [inline src/client/datascience/notebookStorage/vscNotebookModel.ts:103]: In one of @DavidKutu PR i noticed this failing.
Ignore cells that have been closed/deleted.

File: src/test/datascience/kernelLauncher.vscode.test.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ suite('DataScience - Kernel Launcher', () => {
         language: 'python',
         display_name: 'Python 3',
         metadata: {},
-        argv: ['python', '-m', 'ipykernel_launcher', '-f', `{connection_file}`],
+        argv: [PYTHON_PATH, '-m', 'ipykernel_launcher', '-f', `{connection_file}`],
```
Comment (DonJayamanne) [inline src/test/datascience/kernelLauncher.vscode.test.ts:33]: Found a weird issue, an error with ipykernel missing, i guess `python` was pointing to some other interpreter, either way this is a safe change (always use the designated interpreter)

**Review by rchiodo — APPROVED** [review]

**Pull request merged**